### PR TITLE
Added Matrix Exponentiation

### LINKED ISSUE #1180: Adding Matrix Exponentiation
I was just browsing through  the algorithms and data structures but couldn't find Matrix Exponentiation, which is useful for solving linear recurrences in logarithmic time. So, is this something you guys think should be added? If yes then I can work on it and submit a PR. Thanks

### PR DESCRIPTION
**Pull request by anirudnits**
Fixes #1180  Added the **matrix_exponentiation.py** file in **maths directory**

### CODE REVIEW

File: maths/matrix_exponentiation.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+	code = '''
+n = randint(1,70000)
+
+f1 = 1
+f2 = 1
+
+
+fibonacci_with_matrix_exponentiation(n, f1, f2)
+'''
```
Comment (cclauss) [inline maths/matrix_exponentiation.py]: For readability, simplify this to: __code = "fibonacci_with_matrix_exponentiation(randint(1,70000), 1, 1)"

File: maths/matrix_exponentiation.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+f1 = 1
+f2 = 1
+
+
+simple_fibonacci(n, f1, f2)
+'''
+	exec_time = timeit.timeit(setup = setup,
+						  stmt = code,
+						  number = 100)
```
Comment (cclauss) [inline maths/matrix_exponentiation.py]: You might reformat your code with https://github.com/python/black

File: maths/matrix_exponentiation.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+	print ("Without matrix exponentiation the average execution time is ", exec_time/100)
```
Comment (cclauss) [inline maths/matrix_exponentiation.py]: No space between __print__ and __(__.  __print()__ is a function call just like __randint()__ is.

File: maths/matrix_exponentiation.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+	
+def main():
+	matrix_exponentiation_time()
+	simple_fibonacci_time()
+
+
+if __name__ == "__main__":
+	main()
```
Comment (cclauss) [inline maths/matrix_exponentiation.py]: 🚫 GitHub editor is telling you to add a single carriage return at the end of your file.

**Comment by anirudnits** [comment]
Implemented the changes and re-submitted the PR, have a look.

**Pull request merged**